java/pgjdbc_using_dsql_connector/src/main/java/com/amazon/dsql/samples/CustomCredentialsProviderExample.java

```diff
@@ -0,0 +1,191 @@
+                System.out.println("Skipping assume role example - ROLE_ARN environment variable not set");
+                return;
+            }
+            
+            System.out.println("Using role ARN: " + ROLE_ARN);
+            
+            // Create STS client - extract region from cluster endpoint
+            String region = extractRegionFromEndpoint(CLUSTER_ENDPOINT);
+            StsClient stsClient = StsClient.builder()
+                    .region(Region.of(region))
+                    .build();
+            
+            // Create assume role credentials provider
+            AwsCredentialsProvider assumeRoleCredentials = StsAssumeRoleCredentialsProvider.builder()
+                    .refreshRequest(
+                            AssumeRoleRequest.builder()
+                                    .roleArn(ROLE_ARN)
+                                    .roleSessionName("aurora-dsql-session")
+                                    .durationSeconds(3600) // 1 hour
+                                    .build()
+                    )
+                    .stsClient(stsClient)
+                    .build();
+            
+            // Set the custom credentials provider
+            AuroraDsqlCredentialsManager.setProvider(assumeRoleCredentials);
+            
+            // Connect to Aurora DSQL using URL
+            String url = "jdbc:aws-dsql:postgresql://" + CLUSTER_ENDPOINT + "/postgres?user=" + CLUSTER_USER;
+            
+            try (Connection connection = DriverManager.getConnection(url)) {
+                System.out.println("✓ Connected using assume role credentials");
+                executeSimpleQuery(connection);
+            }
+            
+        } catch (Exception e) {
+            System.err.println("✗ Assume role credentials connection failed: " + e.getMessage());
+            e.printStackTrace();
+        }
+    }
+    
+    /**
+     * Extract region from Aurora DSQL endpoint
+     */
+    private static String extractRegionFromEndpoint(String endpoint) {
+        if (endpoint == null || endpoint.isEmpty()) {
+            return "us-east-1"; // default fallback
+        }
+        
+        // Extract region from endpoint like: cluster-id.dsql.us-east-1.on.aws
+        String[] parts = endpoint.split("\\.");
+        if (parts.length >= 3 && parts[2] != null) {
+            return parts[2]; // us-east-1
+        }
+        return "us-east-1"; // default fallback
+    }
+    
+    /**
+     * Execute a simple query to verify the connection
+     */
+     private static void executeSimpleQuery(Connection connection) {
+        if (connection == null) {
+            System.err.println("Connection is null, cannot execute query");
+            return;
+        }
+        
+        try (Statement stmt = connection.createStatement()) {
+            
+            // Test current user
+            try (ResultSet rs = stmt.executeQuery("SELECT current_user")) {
```

Comment:
Are all 3 of these queries necessary? There's nothing inherently wrong with running all 3, but given we are just confirming we can connect to the database it seems like 1 query would be sufficient and would reduce noise for users reading this.